internal/impl/kafka/enterprise/redpanda_migrator_offsets_output.go

```diff
@@ -265,6 +317,36 @@ func (w *redpandaMigratorOffsetsWriter) Write(ctx context.Context, msg *service.
 					delete(partitionOffsets, partition)
 				}
 				offset.Metadata = offsetMetadata
+
+				// As an optimisation to try and avoid unnecessary duplicates in the common case, we check if the
```

Comment:
were you able to write tests for these cases or is that too complicated?

Reply:
I have to clean them up, but yeah, I have a few that are reproducible. I'll try to push them in the next few days.

Reply:
OK, I was able to push some tests to cover this logic. They cover the happy flows, but I can come up with some error cases as well if needed. They're a bit more annoying to assert though.